src/components/ledger/LedgerConnector.tsx

```diff
@@ -98,7 +102,12 @@ export function LedgerConnector({
       addressList: AddressType[] = []
     ) => {
       const address = getAddressFromXPub(xpubValue, accountIndex);
+
       const { balance } = await getAvaxBalance(address);
+
+      if (balance.balanceDisplayValue && balance.balanceDisplayValue !== '0') {
+        lastAccountIndexWithBalance.current = accountIndex;
+      }
```

Comment:
This only works if `getAddressFromXpubKey()` is always called in the correct order (which can be easily broken and would result in a bug hard to fix).

We should check if `accountIndex` truly is greater than `lastAccountIndexWithBalance.current`.

**EDIT:**
I just noticed we add `balance` in `setAddresses(...)` call just below. Would it not be easier to get the `lastAccountIndexWithBalance` by doing something like this:

```ts
// have to do Math.max(), because findLastIndex() may return -1
const lastIndexWithBalance = Math.max( 
  0,
  addressList.findLastIndex((addr) => addr.balance !== '0'),
);
```

❓